### ISSUE TITLE
llext: Better memory management and mmu/mpu support

### ISSUE THREAD

**Issue by teburd** [zephyrproject-rtos/zephyr#63535]
**Is your enhancement proposal related to a problem? Please describe.**
Systems with memory protection schemes such as the arm mpu or an mmu with permission bits require
some support in llext for elf section placement and permissions.

The simplified single heap allocation and copy scheme in the initial PR were good to get started with but are very limiting.

**Describe the solution you'd like**
ELF sections have correct permissions set on the memory they reside in

.text => RX
.data => RW
.rodata => RO
.bss => RW

So perhaps llext_loader is setup a set of callback that takes an elf section and produces a pointer to that sections data with the appropriate permissions setup. This could be viewed as an allocator+copier+permissions setting call in one.

```c
struct llext_loader {
    void *(*section_init)(struct llext_loader *ldr, struct elf_shdr_t shdr);
}
```

**Comment by teburd** [comment]
Fixed by #67857

**Comment by 3Nigma** [comment]
> Fixed by #67857

Can this be closed, then?

**Comment by teburd** [comment]
Yes :-)
